react/tic-tac-toe/src/app/index.js

```diff
@@ -1,11 +1,46 @@
-import React from 'react';
-
+import React, { Component } from 'react';
+import { Route, Switch, BrowserRouter } from 'react-router-dom';
 import Game from './screens/Game';
-
+import Login from './screens/Login';
+import api from '../config/api';
+import PrivateRoute from './components/PrivateRoute';
+import { connect } from 'react-redux';
+import actionsCreators from '../redux/login/actions';
 import '../scss/application.scss';
 
-function App() {
-  return <Game />;
+class App extends Component{
+  componentDidMount() {
+    var token = window.localStorage.getItem('token');
+    if (token) {
+      api.setHeader({Authorization: token});
+      this.props.setAuthenticated(true);
+      this.props.setToken(token)
+    }
+  }
+  render() {
+    return (
+    <BrowserRouter>
+      <Switch>
+        <Route path="/login" render={(routeProps) => (<Login {...routeProps} userAuthenticated={this.props.userAuthenticated} loading={this.props.loading} hasError={this.props.error} checkCredentials={this.props.checkCredentials} />)} />
```

Comment:
What about creating a `PublicRoute` component similar to `PrivateRoute` to wrap the `Login` and avoid defining it inside de `render` prop.